### ISSUE TITLE
[FEATURE] Introudce `doc_cfg` for annotating items that require crate-features

### ISSUE THREAD

**Issue by Cryptjar** [gyscos/cursive#574]
**Is your feature request related to a problem? Please describe.**
Me, as a _no-default-featuers_ user (i.e. using other backends), I always wondered why the `ncurses` backend got special treatment when it comes to constructor functions. So for instance in the [docs][fucntions] you find a function to specifically create a `ncurses`-backed instance, but not for the other backends. So I was forced to simply use the `default` function and hope it will select the right backend (easy if you just want to support a single backend).

That is specifically, until I came across the [`cursive::CursiveExt`][cursive-ext] trait, which, at least in the example, revealed that there will be alternative functions if the respective backends are activated. Tho, in my opinion, it looks extremely weird that the trait definition only has the one function (in the docs).

**Describe the solution you'd like**
Maybe, I'm already a bit spoiled when it comes to [`doc-cfg`] (notice, it works also very well for crate-features). But, I really think the documentation (at least on `docs.rs`) would really profit from this feature. However, since it is nightly, I would stuff it behind a crate-feature such as `doc-cfg` (or whatever).

Concretely, I would add said crate-feature to the `Cargo.toml` e.g.:
```toml
[features]
doc-cfg = [] # Only for fancier docs
```

And replace:
```toml
[package.metadata.docs.rs]
features = ["unstable_scroll", "markdown", "toml"]
```
with
```toml
[package.metadata.docs.rs]
all-features = true # Just document everything on docs.rs
```
Then add the gated feature gate at the top of `lib.rs` :
```rust
// Enable annotating features requirements in docs
#![cfg_attr(feature = "doc-cfg", feature(doc_cfg))]
```
And add also a gated `doc_cfg` to each `cfg`-ed function e.g.:
```rust
/// Creates a new Cursive root using a termion backend.
#[cfg(feature = "termion-backend")]
#[cfg_attr(feature = "doc-cfg", doc(cfg(feature = "termion-backend")))]
pub fn termion() -> CursiveRunnable {
    CursiveRunnable::termion()
}
```

**Describe alternatives you've considered**
Maybe just adding some prose to the docs describing how to handle alternative backends (i.e. not `ncurses`), and mention the existence of the undocumented functions.

[fucntions]: https://docs.rs/cursive/0.16.3/cursive/#functions
[cursive-ext]: https://docs.rs/cursive/0.16.3/cursive/trait.CursiveExt.html#examples
[`doc-cfg`]: https://doc.rust-lang.org/unstable-book/language-features/doc-cfg.html

**Comment by gyscos** [comment]
Hi, and thanks for the report!

Indeed, setting `all-features = true` for the docs is probably a good idea. I wasn't using `doc_cfg` so far because of the nightly requirement, but I didn't think of only enabling it on docs.rs - smart!

**Comment by Cryptjar** [comment]
It appears that this has been implemented by eb9f1b1a81d3fd4c94b283f0730af432448662a5 which is on crates.io since v0.17.0 (January 2022). Thanks.
